crates/sui-json-rpc-types/src/lib.rs

```diff
@@ -51,6 +56,209 @@ use sui_types::sui_serde::{Base64, Encoding};
 mod rpc_types_tests;
 
 pub type GatewayTxSeqNumber = u64;
+pub type SuiMoveTypeParameterIndex = u16;
```

Comment:
We should be able to serialize to a string value

Reply:
yes, but is it any good to do so, TypeParameterIndex is u16 on Move side?

Reply:
Sorry I was confused about how should the client interpret this number. Do you know?

Reply:
from the comment I think it should only be used on the move side based with `FunctionHandle` and `StructHandle`, as on the client side, we only have normalized function and struct, so the index is probably no longer relevant iiuc.

@tnowacki any wisdom? thanks!
```
/// Type parameters are encoded as indices. This index can also be used to lookup the kind of a
/// type parameter in the `FunctionHandle` and `StructHandle`.
```

Reply:
Sorry, what are you trying to do here? I feel like I'm missing some context

Reply:
@tnowacki thanks for the quick reply! We were wondering if `TypeParameterIndex` is still useful from the client side, iiuc it is useful in the context of FunctionHandle and StructHandle, but not any more in the context of normalized module & function, so from the client side, we should just ignore it, did I miss anything?

Reply:
I think it is really required. And IIRC the whole json side of things ignores generics right now (which might be a problem)

If you have something like `entry fun foo<T: drop, U: store>(x: T, y: U)` 
at the compiled module level you will have something like 
```
FunctionHandle {
  is_entry: true,
  type_arguments: vec![Drop, Store],
  arguments: Signature(/* indirect but */vec![TypeParameterIndex(0), TypeParameterIndex(1)]),
}
```
You need to know the constraints on these generics to call the thing correctly. And you need to know how they are used in the parameter positions. 
(similar things for struct handles, as the type will have constraints on its generics)

But I'm not sure how much information you are showing at the client side.